nilearn/plotting/displays.py

```diff
@@ -1255,3 +1260,47 @@ def get_slicer(display_mode):
 def get_projector(display_mode):
     "Internal function to retrieve a projector"
     return get_create_display_fun(display_mode, PROJECTORS)
+
+
+def check_threshold(threshold, data, percentile_calculate, name):
+    """
+    Checks that the given threshold has an accepted string value
+    and returns a threshold computed on the data. Use case of data
+    is mainly from plot_connectome and plot_prob_atlas.
+
+    Parameters
+    ----------
+    threshold: a real value or a percentage in string or list of these.
+        For example, if threshold is a percentage expressed in a string
+        it must finish with a percent sign like "99.7%".
+    data: a numpy array of a map or a numpy array of a symmetric matrix.
+    percentile_calculate: define the name of a specific percentile
```

Comment:
Your code won't work with a string. If you want to handle both `scoreatpercentile` and `fast_abs_percentile`, I would add a `exact=True` parameter rather than passing the function.